cmdstanpy/model.py

```diff
@@ -690,24 +676,25 @@ def sample(
                             dynamic_ncols = False
                         else:
                             dynamic_ncols = True
-                        pbar = [
-                            # warmup
-                            tqdm_pbar(
+
+                        # Total number of iterations (warmup + sample)
+                        # 2000 is default in cmdstan
+                        total_iters = 2000
```

Comment:
Will this work with non default values?

Reply:
Oops, it did work when both the number of warmup and sampling iterations were omitted by the user. But it did not work if only one of them was omitted.

Fixed here:
https://github.com/stan-dev/cmdstanpy/pull/197/commits/79169bc0e7c5230852749f9bf27bb9977c62ce01

Reply:
Nah, this is better

https://github.com/stan-dev/cmdstanpy/pull/197/commits/749ed90275081b99a98728ef8d82a5d726c63e44

Stan outputs 12 / 12345, so we know the total number of iterations is 12345. No need to guess that. This is how it worked before, I think.